app/assets/javascripts/discourse/tests/unit/lib/uploads-test.js

```diff
@@ -312,4 +313,57 @@ discourseModule("Unit | Utility | uploads", function () {
       "![image|100x200](/uploads/123/abcdef.ext)"
     );
   });
+
+  test("displayErrorForUpload - jquery file upload - jqXHR present", function (assert) {
+    sinon.stub(bootbox, "alert");
+    displayErrorForUpload(
+      {
+        jqXHR: { status: 422, responseJSON: { message: "upload failed" } },
+      },
+      { max_attachment_size_kb: 1024, max_image_size_kb: 1024 },
+      "test.png"
+    );
+    assert.ok(bootbox.alert.calledWith("upload failed"), "the alert is called");
+  });
+
+  test("displayErrorForUpload - jquery file upload - jqXHR missing, errors present", function (assert) {
+    sinon.stub(bootbox, "alert");
+    displayErrorForUpload(
+      {
+        errors: ["upload failed"],
+      },
+      { max_attachment_size_kb: 1024, max_image_size_kb: 1024 },
+      "test.png"
+    );
+    assert.ok(bootbox.alert.calledWith("upload failed"), "the alert is called");
+  });
+
+  test("displayErrorForUpload - jquery file upload - no errors", function (assert) {
+    sinon.stub(bootbox, "alert");
+    displayErrorForUpload(
+      {},
+      {
+        max_attachment_size_kb: 1024,
+        max_image_size_kb: 1024,
+      },
+      "test.png"
+    );
+    assert.ok(
+      bootbox.alert.calledWith(I18n.t("post.errors.upload")),
+      "the alert is called"
```

Comment:
Not sure if I understand this test. There are not errors present but the bootbox alert will still be called?

Reply:
Yes this is a valid case that I kept from the original code for displayUploadError:

https://github.com/discourse/discourse/pull/13787/files/0da1e94bdde85caacab9fea71a969dae9b1e3c56#diff-6422168f859b8d76bc3dbfafe111ffd8e4178a96c01996d53b617c8616b1345cL297

I think it's for cases where there is an error for the promise that is run through `catch`, but does not have any of the information from the server side

Reply:
Every time `displayUploadError` is called it's called via `catch`

Reply:
ahh icic got it :+1: